tests/index.test.js

```diff
@@ -808,6 +813,164 @@ describe('Red Perfume', () => {
               .not.toHaveBeenCalled();
           });
         });
+
+        describe('Qualifying', () => {
+          const qualifyingCSS = `
+            h1.qualifying {
+              border: 1px solid #000;
+              padding: 10px;
+              line-height: 1.4;
+            }
+          `;
+
+          test('Normal', () => {
+            options = {
+              verbose: true,
+              customLogger: jest.fn(),
+              tasks: [
+                {
+                  uglify: false,
+                  styles: {
+                    data: qualifyingCSS,
+                    result: function (result, err) {
+                      const expectation = testHelpers.trimIndentation(`
+                        h1.rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 {
+                          border: 1px solid #000;
+                        }
+                        h1.rp__padding__--COLON10px {
+                          padding: 10px;
+                        }
+                        h1.rp__line-height__--COLON1__--DOT4 {
+                          line-height: 1.4;
+                        }
+                      `, 24);
+
+                      expect(result)
+                        .toEqual(expectation, undefined);
+
+                      expect(err)
+                        .toEqual(undefined);
+                    }
+                  },
+                  markup: [
+                    {
+                      data: inputMarkup,
+                      result: function (result, err) {
+                        expect(testHelpers.trimIndentation(result))
+                          .toEqual(testHelpers.trimIndentation(`
+                            <!DOCTYPE html><html><head></head><body>
+                              <h1 class="rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 rp__padding__--COLON10px rp__line-height__--COLON1__--DOT4"></h1>
+                              <div class="simple pseudo"></div>
+                              <div class="after">
+                                <div class="nested"></div>
+                              </div>
+                            </body></html>
+                          `, 28));
+
+                        expect(err)
+                          .toEqual(undefined);
+                      }
+                    }
+                  ],
+                  scripts: {
+                    result: function (result, err) {
```

Comment:
> "I'm wondering why the test for the script tag on line 876 in `index.test.js` is `.qualifying`. Shouldn't it be `h1.qualifying`?"

@gwenf Don't add PR comments, add inline comments, that can be replied to directly, like this.


The JSON output is meant to be used by JavaScript, so you could do something like this:
```html
<div :class="$styles.qualifying"></div>
<h1 :class="$styles.qualifying"></h1>
```
and it would show up in the DOM as:
```html
<div class="rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 rp__padding__--COLON10px rp__line-height__--COLON1__--DOT4"></div>
<h1 class="rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 rp__padding__--COLON10px rp__line-height__--COLON1__--DOT4"></h1>
```

But the CSS that actually applies the styles would be:

```css
h1.rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 { border: 1px solid #000 }
h1.rp__padding__--COLON10px { padding: 10px }
h1.rp__line-height__--COLON1__--DOT4 { line-height: 1.4 }
```

Or something to that effect, the JS portion will change later. Depending on edge cases as we go further, we may need to change these qualifying classes to include the tag name. Like:

```css
h1.h1-rp__padding__--COLON10px { padding: 10px }
```

Reply:
Oh, I see now. Thanks.